open_schools_platform/common/services.py

```diff
@@ -172,9 +173,14 @@ def __init__(self):
         self.email_transport = EMAIL_TRANSPORT
 
 
-def exception_if_email_service_unavailable():
-    if SendEmailService().email_transport is None:
-        raise EmailServiceUnavailable()
+class email_service:
+    def __enter__(self):
+        if CommonConstants.REGISTRATION_MESSAGES_TRANSPORT == "email" and \
+                SendEmailService().email_transport is None:
+            raise EmailServiceUnavailable()
+
+    def __exit__(self, type, value, traceback):
+        pass
```

Comment:
Seems good, but why don't we just add a new condition in previous functionality?